### ISSUE TITLE
(RFC) Filter options

### ISSUE THREAD

**Issue by dolanske** [Mavulp/hivefriends#16]
To add filters to album lists, front-end must know on which values to filter.

For example to filter on users, I need something like this.

```js
const options = {
    // Key must match the path parameter to use in filters
    username: ['dolanske','tmtu','Jokler','kilmanio']
    ...
}
```

Getting these values on front-end could mean a lot of nested loops and possibly cause issues on slower machines when creating filter options in large datasets. Eg.: list of all albums, which could house hundreds of albums, where each album contains multiple values to filter.

**Question 1**: Do we want to offload this logic on backend and possibly cache it? 

**Question 2**: Which values do we allow to filter on? For starters I'll name a few

1. Users (multiple)
2. Upload date 
3. Time range `from` & `to`

For instance if we only set `from`, `to` should automatically be set to `now`. And if only `to` is set it should automatically pick the oldest `from` from available abums.

**Comment by thechantaro** [comment]
Chantaro

**Comment by dolanske** [comment]
Resolved by #31
